src/main/scala/com/microsoft/sqlserver/jdbc/spark/utils/BulkCopyUtils.scala

```diff
@@ -239,53 +267,85 @@ object BulkCopyUtils extends Logging {
         val dfCols = df.schema
 
         val tableCols = getSchema(rs, JdbcDialects.get(url))
+        val computedCols = getComputedCols(conn, dbtable)
+        var dfComputedCols : List[String] = List()
+
         val prefix = "Spark Dataframe and SQL Server table have differing"
 
-        assertCondition(dfCols.length == tableCols.length,
-            s"${prefix} numbers of columns")
+        if (computedCols.length == 0) {
+            assertCondition(dfCols.length == tableCols.length,
+                s"${prefix} numbers of columns")
+        }
+        else {
```

Comment:
Not clear why this detailed check. The data frame may or may not specify the computed column. It may specify the computed column without saying its computed. There is no indicator available to do a length check.

Reply:
I used col_count_tbl == col_count_df - computed_col_count_df + computed_col_tbl for length check if tbl has computed cols. As we discussed, it is redundant. Now it is removed.

Reply:
I removed the check col_count_tbl == col_count_df - computed_col_count_df + computed_col_tbl.
But there is one issue: e.g. when tbl has three cols: 
    Id bigint,
    Year nvarchar(4),
    Year_1 AS (Year + '-' + Year), 
while df has three cols: "Id", "Year", "Month", it will append successful even if Month col does not exist in tbl.

Reply:
> I removed the check col_count_tbl == col_count_df - computed_col_count_df + computed_col_tbl for length check if tbl has computed cols..
> But there is one issue: e.g. when tbl has three cols:
> Id bigint,
> Year nvarchar(4),
> Year_1 AS (Year + '-' + Year),
> while df has three cols: "Id", "Year", "Month", it will append successful even if Month col does not exist in tbl.

Ok. You may put the length check under the strictSchemaCheck. Please simplify the check if you do so. Right now, its hard to understand.

IMO this is a good feature to have when a partial dataframe can be used to write to a SQL table.  so when strictSchemaCheck is False we should allow for such a write to happen.